src/pages/home/ReportScreen.tsx

```diff
@@ -14,154 +17,116 @@ import OfflineWithFeedback from '@components/OfflineWithFeedback';
+import type * as OnyxTypes from '@src/types/onyx';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
 import HeaderView from './HeaderView';
-import reportActionPropTypes from './report/reportActionPropTypes';
 import ReportActionsView from './report/ReportActionsView';
 import ReportFooter from './report/ReportFooter';
 import {ActionListContext, ReactionListContext} from './ReportScreenContext';
+import type {ActionListContextType, ReactionListRef, ScrollPosition} from './ReportScreenContext';
 
-const propTypes = {
-    /** Navigation route context info provided by react navigation */
-    route: PropTypes.shape({
-        /** Route specific parameters used on this screen */
-        params: PropTypes.shape({
-            /** The ID of the report this screen should display */
-            reportID: PropTypes.string,
-
-            /** The reportActionID to scroll to */
-            reportActionID: PropTypes.string,
-        }).isRequired,
-    }).isRequired,
-
+type ReportScreenOnyxProps = {
     /** Tells us if the sidebar has rendered */
-    isSidebarLoaded: PropTypes.bool,
+    isSidebarLoaded: OnyxEntry<boolean>;
 
-    /** The report currently being looked at */
-    report: reportPropTypes,
+    /** Beta features list */
+    betas: OnyxEntry<OnyxTypes.Beta[]>;
 
-    /** The report metadata loading states */
-    reportMetadata: reportMetadataPropTypes,
+    /** The policies which the user has access to */
+    policies: OnyxCollection<OnyxTypes.Policy>;
 
-    /** All the report actions for this report */
-    reportActions: PropTypes.objectOf(PropTypes.shape(reportActionPropTypes)),
+    /** The account manager report ID */
+    accountManagerReportID: OnyxEntry<string>;
 
-    /** The report's parentReportAction */
-    parentReportAction: PropTypes.shape(reportActionPropTypes),
+    /** Whether user is leaving the current report */
+    userLeavingStatus: OnyxEntry<boolean>;
 
     /** Whether the composer is full size */
-    isComposerFullSize: PropTypes.bool,
+    isComposerFullSize: OnyxEntry<boolean>;
 
-    /** Beta features list */
-    betas: PropTypes.arrayOf(PropTypes.string),
+    /** All the report actions for this report */
+    reportActions: OnyxTypes.ReportAction[];
 
-    /** The policies which the user has access to */
-    policies: PropTypes.objectOf(
-        PropTypes.shape({
-            /** The policy name */
-            name: PropTypes.string,
+    /** The report currently being looked at */
+    report: OnyxEntry<ReportWithoutHasDraft>;
 
-            /** The type of the policy */
-            type: PropTypes.string,
-        }),
-    ),
+    /** The report metadata loading states */
+    reportMetadata: OnyxEntry<OnyxTypes.ReportMetadata>;
 
-    /** The account manager report ID */
-    accountManagerReportID: PropTypes.string,
+    /** The report's parentReportAction */
+    parentReportAction: OnyxEntry<OnyxTypes.ReportAction>;
+};
 
+type OnyxHOCProps = {
```

Comment:
Why do we have a separate type for this prop? Couldn't be inside `ReportScreenProps`?

```ts
type ReportScreenProps = ViewportOffsetTopProps & CurrentReportIDContextValue & ReportScreenOnyxProps & ReportScreenNavigationProps & {
    /** Onyx function that marks the component ready for hydration */
    markReadyForHydration?: () => void;
};
```

Reply:
This way I wanted to point out that this function goes from the [withOnyx HOC](https://github.com/Expensify/react-native-onyx/blob/main/lib/withOnyx.js#L327).
Overall I guess this type should be on the Onyx side, and withOnyx HOC typing should be updated accordingly

Reply:
Yes you are right, I think we can declare a `WithOnyxProps` type in `lib/withOnyx.d.ts` and omit it together with `TOnyxProps`.

```tsx
type WithOnyxProps = {
    markReadyForHydration: () => void;
};

declare function withOnyx<TComponentProps, TOnyxProps>(
    mapping: {
        [TOnyxProp in keyof TOnyxProps]: OnyxPropMapping<TComponentProps, TOnyxProps, TOnyxProp> | OnyxPropCollectionMapping<TComponentProps, TOnyxProps, TOnyxProp>;
    },
    shouldDelayUpdates?: boolean,
): (component: React.ComponentType<TComponentProps>) => React.ComponentType<Omit<TComponentProps, keyof (TOnyxProps & WithOnyxProps)>>;

export {WithOnyxProps}; // import and export in `lib/index.d.ts` too.
```

So if someone needs to use `markReadyForHydration` in its component they need to import `WithOnyxProps` and include into the component props

```tsx
type TestProps = TestOnyxProps &
    WithOnyxProps & {
        id: string;
    };
```

wdyt?

Reply:
After testing of withOnyx HOC typing updates, it turned out it will require app side type updates in the app HOCs which uses withOnyx.

For example `withPolicy` HOC return type will need to be updated
from
`React.ComponentType<Omit<TProps, keyof WithPolicyOnyxProps>>`
to
`React.ComponentType<Omit<TProps, keyof (WithPolicyOnyxProps & WithOnyxProps)>> `

But in more complex HOCs, which use other HOCs the typing becomes too complex and less readable, for example:
![image](https://github.com/Expensify/App/assets/23176449/b487f3ee-6b58-4506-b182-8dcdc21aef8e)
Though I was able to resolve all caused TS issues, at least at this stage.

But since `markReadyForHydration` function is used on the Report screen only, maybe it makes sense to leave it as is, to escape HOCs typing complexity.

@fabioh8010 @blazejkustra what do you think?

Reply:
I agree, let's not overcomplicate this